patterns.md

```diff
@@ -0,0 +1,63 @@
+# Constants in patterns
+
+Constants that pass the [structural equality check](https://github.com/rust-lang/rust/issues/74446) can be used in patterns:
+```rust
+const FOO: i32 = 13;
+
+fn is_foo(x: i32) -> bool {
+  match x {
+    FOO => true,
+    _ => false,
+  }
+}
+```
+However, that check has some loopholes, so e.g. `&T` can be used in a pattern no matter the `T`.
+
+A const-pattern is compiled by calling `PartialEq::eq` to compare the subject of the `match` with the constant,
```

Comment:
Is it really always `PartialEq`? I would imagine that e.g.
```rust
const TEST: (i32, i32) = (0, 0);

match (x, y) {
    (z1, z2) if z1 != z2 => ...,
    TEST => ....
    (z, _) => ...
}
```
compiles the same as when `TEST` is inlined?

Reply:
We are taking it apart, but we should not. All constants should be treated as opaque according to the status quo of what has been RFCed

Reply:
@oli-obk  could you propose something concrete we could write for the remaining TODO?

Reply:
I wrote something up. Should we also mention that this diverges from the RFCed semantics?

Reply:
:laughing: ok you had the same thought^^